src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -82,13 +86,38 @@ public function process(ContainerBuilder $container)
                         $event['method'] = '__invoke';
                     }
                 }
+                $registeredMethods[$event['method']] = true;
 
                 $definition->addMethodCall('addListener', [$event['event'], [new ServiceClosureArgument(new Reference($id)), $event['method']], $priority]);
-
                 if (isset($this->hotPathEvents[$event['event']])) {
                     $container->getDefinition($id)->addTag($this->hotPathTagName);
                 }
             }
+
+            if ($isAutoregisterEnabled) {
+                if (null === ($class = $container->getDefinition($id)->getClass()) || null === ($r = $container->getReflectionClass($class, false))) {
```

Comment:
I think you don't need the part before `||`

Reply:
same code copy/pasted from line 87. Should I change it too?